### PR TITLE
Patch ntapi more thoroughly

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

We needed more tender and loving care for patching `ntapi`, still breaking windows build after #31961 

#### Summary of Changes

patch moar!

(i'll create a proper pr at the spl-token repo...)

Fixes #

### CODE REVIEW

**Review by yihau — APPROVED** [review]
can be the stopgap. I think maybe we can remove spl-token-cli from cargo-install-all in the future 🙈

**Comment by ryoqun** [comment]
(I'll merge this as soon as check passes before waiting full ci run; technically master ci is broken right now....)

File: scripts/cargo-install-all.sh [hunk 1]
```diff
@@ -142,8 +142,12 @@ mkdir -p "$installDir/bin"
 
   # Exclude `spl-token` binary for net.sh builds
   if [[ -z "$validatorOnly" ]]; then
+    # the config is work around until we ship newer spl-token-cli...
     # shellcheck disable=SC2086 # Don't want to double quote $rust_version
-    "$cargo" $maybeRustVersion install --locked spl-token-cli --root "$installDir"
+    "$cargo" $maybeRustVersion \
+      --config 'patch.crates-io.ntapi.git="https://github.com/solana-labs/ntapi"' \
+      --config 'patch.crates-io.ntapi.rev="5980bbab2e0501a8100eb88c12222d664ccb3a0a"' \
+      install --locked spl-token-cli --root "$installDir"
```
Comment (ryoqun) [inline scripts/cargo-install-all.sh:150]: here

Comment (ryoqun) [reply scripts/cargo-install-all.sh:150]: note that we needed to remove --locked as well: #31971

**Pull request merged**